hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -179,14 +189,31 @@ refuelIfNeeded ::
   Actor ->
   Lovelace ->
   IO ()
-refuelIfNeeded tracer node actor amount = do
-  (actorVk, _) <- keysFor actor
-  (fuelUTxO, otherUTxO) <- queryMarkedUTxO node actorVk
-  traceWith tracer $ StartingFunds{actor = actorName actor, fuelUTxO, otherUTxO}
+refuelIfNeeded tracer node receiver amount = do
+  (receivingVk, _) <- keysFor receiver
+  (senderVk, senderSk) <- keysFor Faucet
+  (fuelUTxO, otherUTxO) <- queryMarkedUTxO node receivingVk
+  traceWith tracer $ StartingFunds{actor = actorName receiver, fuelUTxO, otherUTxO}
   let fuelBalance = selectLovelace $ balance @Tx fuelUTxO
   when (fuelBalance < amount) $ do
-    utxo <- seedFromFaucet node actorVk amount Fuel (contramap FromFaucet tracer)
-    traceWith tracer $ RefueledFunds{actor = actorName actor, refuelingAmount = amount, fuelUTxO = utxo}
+    utxo <- sendFundsTo node (senderVk, senderSk) receivingVk amount Fuel (contramap FromFaucet tracer) True
+    traceWith tracer $ RefueledFunds{actor = actorName receiver, refuelingAmount = amount, fuelUTxO = utxo}
+
+-- | Return the remaining funds to the faucet
+returnAssetsToFaucet ::
+  Tracer IO EndToEndLog ->
+  RunningNode ->
+  Actor ->
+  IO ()
+returnAssetsToFaucet tracer node@RunningNode{networkId, nodeSocket} sender = do
+  (receivingVk, _) <- keysFor Faucet
+  (senderVk, senderSk) <- keysFor sender
+  utxo <- queryUTxOFor networkId nodeSocket QueryTip senderVk
+  -- Bit ugly bit we need to subtract the fees manually here.
+  -- TODO: Implement the fee calculation for our smoke-tests
+  let returnBalance = (selectLovelace $ balance @Tx utxo) - 1_200_000
+  void $ sendFundsTo node (senderVk, senderSk) receivingVk returnBalance Normal (contramap FromFaucet tracer) False
```

Comment:
Should: do what `sendFundsTo ... Normal ... False` means directly here. 

This is the only usage of `sendFundsTo` with this parameterization, so there is no point in creating a function, which even contains optional semantics (smelly).